approaches/data-modeling/README.md

```diff
@@ -0,0 +1,411 @@
+
+The `type` fields included below show [a subset](https://json-schema.org/understanding-json-schema/reference/type.html) of types available - `string`, `number`, `integer`, `boolean` - to demonstrate the range that's possible with a JSON Schema data model.
+
+<details>
+<summary>Types</summary>
+
+```json
+"properties": {
+  "officeId": {
+    "description": "Unique identifier of the office",
+    "type": {
+      "$ref" : "#/definitions/OfficeId"
+    }
+  },
+  "officeAddress": {
+    "description": "Address of the office",
+    "type": {
+      "$ref" : "#/definitions/Address"
+    }
+  },
+  "maxOccupancy": {
+    "description": "Maximum number of employees allowed",
+    "type": "integer"
+  },
+  "costType" : {
+    "description": "Cost type for the office",
+    "type": "string",
+    "enum": ["LEASE", "MORTGAGE", "OWN", "OTHER"]
+  },
+  "monthlyCost": {
+    "description": "Cost type for the office per month, if applicable",
+    "type": "number"
+  },
+  "dateOpened": {
+    "description": "Date the office opened",
+    "type": "string",
+    "format": "date"
+  },
+  "isFull": {
+    "description": "Whether the office is at its max capacity",
+    "type": "boolean"
+  }
+}
+```
+</details>
+
+The types selected for each field cannot be modified after a Uni is created so they are especially important to get right from the start.
+
+You can use the GraphQL Explorer or Entity Explorer views of the Vendia Share web app to review which JSON Schema types from the [data model](uni_configuration/schema.json) resulting in which GraphQL Schema types.
+
+### Approach 5 - Leverage Formats
+When using the `string` type, you may also want to apply one of the pre-defined `format` values [supported by JSON Schema](https://json-schema.org/understanding-json-schema/reference/string.html#built-in-formats).  This not only makes the schema more understandable by your partners (who may be sharing data with you) but also adds another layer of syntax validation for all GraphQL mutations.
+
+For example, the `Office` entity's `dateOpened` field has a `format` set to `date`.  Any mutation on the `Office` entity that includes a `dateOpened` field will have its value for `dateOpened`  validated against the JSON Schema `date` format of `yyyy-mm-dd`.  Any value that fails that validation check will immediately result in a failed mutation operation for a GraphQL client. 
+
+<details>
+<summary>Formats</summary>
+
+```json
+"dateOpened": {
+  "description": "Date the office opened",
+  "type": "string",
+  "format": "date"
+}
+```
+</details>
+
+The formats available to further restrict `string` fields are certainly helpful, but sometimes a custom format is needed.
+
+### Approach 6 - Leverage Patterns
```

Comment:
Within either patterns or formats, it might be worth describing the use of enum for a set of allowed values, as the demo example uses that technique

Reply:
Added